extmod/nimble/modbluetooth_nimble.c

```diff
@@ -763,6 +763,8 @@ STATIC int peripheral_discover_service_cb(uint16_t conn_handle, const struct ble
     if (error->status == 0) {
         mp_obj_bluetooth_uuid_t service_uuid = create_mp_uuid(&service->uuid);
         mp_bluetooth_gattc_on_primary_service_result(conn_handle, service->start_handle, service->end_handle, &service_uuid);
+    } else {
+        mp_bluetooth_gattc_on_discover_complete(MP_BLUETOOTH_IRQ_GATTC_SERVICES_COMPLETE, conn_handle, error->status);
     }
```

Comment:
What's you opinion on the idea of reporting 0 instead of `BLE_HS_EDONE`? Thereby we could state in the API doc that "status is 0 if the discover operation completed successfully, or an implementation-specific error code otherwise".

Reply:
Yes, I'm inclined to agree, good suggestion. Will fix.

Reply:
Done, and confirmed that this behaviour matches read/write status (and same for btstack).